### DISCUSSION TITLE
Only Show X-Axis Labels for Data Points (Piggy Back on #509)

### DISCUSSION THREAD

**Discussion by sean-brown-dev** [patrykandpatrick/vico#510] [Questions]
This is basically the same question as #509, but a bit different. I have unevenly spaced x-values as well so I get labels on the x-axis with no data point, which is the expected default behavior. Is it possible to make the chart only show x-axis labels for actual data points?

For example I have a data point at 12/18 and 12/21. Can I have it only show 12/18 and 12/21 instead of showing 12/18, 12/19, 12/20 and 12/21?

EDIT: The reason I was asking this was because I was having an issue in v13.0.1 but it no longer happens in alpha, so no need to answer.

**Comment by patrickmichalik** [comment]
Hello! I’m glad to hear that Vico 2 resolves the issue you were experiencing in version 1.13.1. In case someone has the same question and stumbles upon this discussion, let me answer. To achieve the behavior described, you can create a custom `AxisItemPlacer.Horizontal` implementation. The simplest approach would be to copy over `DefaultHorizontalAxisItemPlacer` and add the required checks. We plan on adding this functionality to the default `AxisItemPlacer.Horizontal` implementation down the line—`AxisItemPlacer.Horizontal.default` will have an extra boolean parameter.
